pandas/core/frame.py

```diff
@@ -8811,18 +8812,18 @@ def append(
 
         Examples
         --------
-        >>> df = pd.DataFrame([[1, 2], [3, 4]], columns=list('AB'))
+        >>> df = pd.DataFrame([[1, 2], [3, 4]], columns=list('AB'), index=[0, 2])
```

Comment:
The comment from @datapythonista finding the original clearer, I agree with, why do you not change the index here to ["x", "y"] and for df2. Then you achieve; highlighting the non-unique element created; avoid any confusion with the same 0,1 and indexes which are only conincidentally the same in the original example; and demonstrate the index has a dtype change rom string to base RangeIndex.

Reply:
Yes, that sounds good!